include/libpmemobj++/experimental/basic_string.hpp

```diff
@@ -1083,7 +1083,7 @@ class basic_string {
 	size_type
 	max_size() const noexcept
 	{
-		return PMEMOBJ_MAX_ALLOC_SIZE / sizeof(CharT);
+		return PMEMOBJ_MAX_ALLOC_SIZE / sizeof(CharT) - sizeof('\0');
```

Comment:
is it really correct? If CharT is 2 bytes you'll need 2 bytes for the terminator, right?